internal/rbac/rbac.go

```diff
@@ -0,0 +1,92 @@
+package rbac
+
+import (
+	"context"
+	"errors"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	kmmv1beta1 "github.com/rh-ecosystem-edge/kernel-module-management/api/v1beta1"
+)
+
+//go:generate mockgen -source=rbac.go -package=rbac -destination=mock_rbac.go
+
+type RBACCreator interface {
+	GetDriverContainerServiceAccount(ctx context.Context, mod kmmv1beta1.Module, sa *corev1.ServiceAccount) error
+	GetDriverContainerRoleBinding(ctx context.Context, mod kmmv1beta1.Module, rb *rbacv1.RoleBinding) error
+	SetDriverContainerServiceAccountAsDesired(ctx context.Context, sa *corev1.ServiceAccount, mod kmmv1beta1.Module) error
+	SetDriverContainerRoleBindingAsDesired(ctx context.Context, rb *rbacv1.RoleBinding, mod kmmv1beta1.Module) error
+}
```

Comment:
Wondering if this interface should be simplified to just:
- `EnsureModuleLoaderRBAC`
- `EnsureDevicePluginRBAC`

This would eliminate the need for `handle*` private methods and reduce the number of calls to mock.

Reply:
Agreed! It smelled funny to me as well :+1: